src/pages/waiting-room-to-car/cat-home-test/waiting-room-to-car.cat-home-test.page.ts

```diff
@@ -120,85 +114,103 @@ export class WaitingRoomToCarCatHomeTestPage extends BasePageComponent {
     public authenticationProvider: AuthenticationProvider,
     public faultCountProvider: FaultCountProvider,
     public questionProvider: QuestionProvider,
+    private testDataByCategoryProvider: TestDataByCategoryProvider,
+    private vehicleDetailsByCategoryProvider: VehicleDetailsByCategoryProvider,
   ) {
     super(platform, navController, authenticationProvider);
-
-    // TODO does this need seperate questions?
-    this.tellMeQuestions = questionProvider.getTellMeQuestions(TestCategory.BE);
     this.form = new FormGroup({});
   }
 
   ngOnInit(): void {
-
     const currentTest$ = this.store$.pipe(
       select(getTests),
       select(getCurrentTest),
     );
 
+    const category$ = currentTest$.pipe(
+      select(getTestCategory),
+      map(category => category as TestCategory),
+    );
+    category$.subscribe((categoryCode) => {
```

Comment:
You need to close every subscription you manually create. I can't see any code added to do that. Each one you don't close creates a memoery leak in the app